### ISSUE TITLE
projection $slice error

### ISSUE THREAD

**Issue by lanlin** [mongodb/mongo-php-library#265]
**Enviroment**
PHP: 7.0.11
Library: 1.0.3
Mongodb: 3.2.9
Extension: mongodb-1.2.0alpha3

When i make a query using the mongo-php-library as this

``` php
$collection->find([], 
    [
        'projection' => [ 'users' => ['$slice' => 10] ]
    ]
);
```

I got an error:
**Cannot use $-modifiers or empty keys in 'opts'.**

Can anyone help me?
Is the '$slice' deperated?

**Comment by jmikola** [comment]
Thanks for the report. This appears to be a bug in libmongoc 1.5.0-rc1, which is used to validate query options. I've opened [CDRIVER-1692](https://jira.mongodb.org/browse/CDRIVER-1692) to track this upstream, so please follow that issue for updates.

**Comment by jmikola** [comment]
@lanlin: [1.2.0](https://github.com/mongodb/mongo-php-driver/releases/tag/1.2.0) of the PHP driver has been released, which includes libmongoc 1.5.0. Please upgrade and report back if the issue is resolved.

The 1.0.3 library version is compatible with the 1.2.0 extension version, so there should be no need to upgrade the library for this issue.
